test/image/mocks/axes_category_ascending.json

```diff
@@ -0,0 +1,13 @@
+{
+    "data": [{
+        "x": ["c","a","e","b","d"],
+        "y": [15,11,12,13,14]}
+    ],
+    "layout": {
+        "xaxis": {
+            "title": "category ascending",
+            "type": "category",
+            "categorymode": "category ascending",
+            "categorylist": ["y","b","x","a","d","z","e","c", "q", "k"]
```

Comment:
**Important note**: the order in which the data points are drawn does not change (e.g. `('c', 15)` is still plotted first followed by `('a', 11)`. Only the locations of the categorical coordinates change - which I think is the correct behavior. Thoughts?